### ISSUE TITLE
Weird notice when running tests via command line

### ISSUE THREAD

**Issue by phyzical** [silverleague/silverstripe-ideannotator#118]
Another small bug not sure if its due to my setup or something on your side.

When i run php vendor/phpunit/phpunit/phpunit it get a notice originating from tests for the IDEAnnotator.... i think... based on the stacktrace below. It hits my test BaseControllerTest but then towards the end it passes through ideannotator tests? 

```
Notice: Array to string conversion in /Users/phyzical/Sites/adsasd/vendor/silverstripe/framework/src/ORM/DataObject.php on line 2102

Call Stack:
    0.0105     399640   1. {main}() /Users/phyzical/Sites/asd/vendor/phpunit/phpunit/phpunit:0
    0.0699    1913944   2. PHPUnit_TextUI_Command::main(???) /Users/phyzical/Sites/asd/vendor/phpunit/phpunit/phpunit:52
    0.0700    1914056   3. PHPUnit_TextUI_Command->run(???, ???) /Users/phyzical/Sites/asd/vendor/phpunit/phpunit/src/TextUI/Command.php:116
    0.3513    5115488   4. PHPUnit_TextUI_TestRunner->doRun(???, ???, ???) /Users/phyzical/Sites/asd/vendor/phpunit/phpunit/src/TextUI/Command.php:186
    0.3626    5362944   5. PHPUnit_Framework_TestSuite->run(???) /Users/phyzical/Sites/asd/vendor/phpunit/phpunit/src/TextUI/TestRunner.php:517
    0.3676    5368896   6. PHPUnit_Framework_TestSuite->run(???) /Users/phyzical/Sites/asd/vendor/phpunit/phpunit/src/Framework/TestSuite.php:733
    4.0076   45042832   7. BaseControllerTest->run(???) /Users/phyzical/Sites/asd/vendor/phpunit/phpunit/src/Framework/TestSuite.php:733
    4.0077   45042832   8. PHPUnit_Framework_TestResult->run(???) /Users/phyzical/Sites/asd/vendor/phpunit/phpunit/src/Framework/TestCase.php:868
    4.0125   45079184   9. BaseControllerTest->runBare() /Users/phyzical/Sites/asd/vendor/phpunit/phpunit/src/Framework/TestResult.php:686
    4.0129   45095776  10. BaseControllerTest->setUp() /Users/phyzical/Sites/asd/vendor/phpunit/phpunit/src/Framework/TestCase.php:909
    4.0129   45095776  11. SilverStripe\Dev\State\SapphireTestState->setUp(???) /Users/phyzical/Sites/asd/vendor/silverstripe/framework/src/Dev/SapphireTest.php:274
    4.0279   45306704  12. SilverStripe\Dev\State\FixtureTestState->setUp(???) /Users/phyzical/Sites/asd/vendor/silverstripe/framework/src/Dev/State/SapphireTestState.php:68
    4.0375   45380552  13. SilverStripe\ORM\Connect\TempDatabase->build() /Users/phyzical/Sites/asd/vendor/silverstripe/framework/src/Dev/State/FixtureTestState.php:44
    4.0399   45380872  14. SilverStripe\ORM\Connect\TempDatabase->resetDBSchema(???) /Users/phyzical/Sites/asd/vendor/silverstripe/framework/src/ORM/Connect/TempDatabase.php:192
    4.0400   45380872  15. SilverStripe\ORM\Connect\TempDatabase->rebuildTables(???) /Users/phyzical/Sites/asd/vendor/silverstripe/framework/src/ORM/Connect/TempDatabase.php:283
    4.0404   45389840  16. SilverStripe\ORM\Connect\MySQLSchemaManager->schemaUpdate(???) /Users/phyzical/Sites/asd/vendor/silverstripe/framework/src/ORM/Connect/TempDatabase.php:246
    4.0409   45389840  17. SilverStripe\ORM\Connect\TempDatabase->SilverStripe\ORM\Connect\{closure}() /Users/phyzical/Sites/asd/vendor/silverstripe/framework/src/ORM/Connect/DBSchemaManager.php:161
    5.6978   50104352  18. singleton(???) /Users/phyzical/Sites/asd/vendor/silverstripe/framework/src/ORM/Connect/TempDatabase.php:230
    5.6978   50104352  19. SilverStripe\Core\Injector\Injector->get(???, ???, ???) /Users/phyzical/Sites/asd/vendor/silverstripe/framework/src/includes/functions.php:34
    5.6978   50104352  20. SilverStripe\Core\Injector\Injector->getNamedService(???, ???, ???) /Users/phyzical/Sites/asd/vendor/silverstripe/framework/src/Core/Injector/Injector.php:946
    5.6979   50104752  21. SilverStripe\Core\Injector\Injector->instantiate(???, ???, ???) /Users/phyzical/Sites/asd/vendor/silverstripe/framework/src/Core/Injector/Injector.php:993
    5.6990   50106632  22. SilverStripe\Core\Injector\Injector->inject(???, ???) /Users/phyzical/Sites/asd/vendor/silverstripe/framework/src/Core/Injector/Injector.php:611
    5.6994   50110936  23. SilverLeague\IDEAnnotator\Tests\SubTeam->__isset(???) /Users/phyzical/Sites/asd/vendor/silverstripe/framework/src/Core/Injector/Injector.php:733
    5.6995   50110976  24. SilverLeague\IDEAnnotator\Tests\SubTeam->hasMethod(???) /Users/phyzical/Sites/asd/vendor/silverstripe/framework/src/View/ViewableData.php:109
    5.6995   50110976  25. SilverLeague\IDEAnnotator\Tests\SubTeam->getExtraMethodConfig(???) /Users/phyzical/Sites/asd/vendor/silverstripe/framework/src/Core/CustomMethods.php:148
    5.6995   50110976  26. SilverLeague\IDEAnnotator\Tests\SubTeam->defineMethods() /Users/phyzical/Sites/asd/vendor/silverstripe/framework/src/Core/CustomMethods.php:161
    5.7008   50165776  27. SilverLeague\IDEAnnotator\Tests\SubTeam->hasMany(???) /Users/phyzical/Sites/asd/vendor/silverstripe/framework/src/ORM/DataObject.php:721
    5.7009   50165776  28. preg_replace(???, ???, ???) /Users/phyzical/Sites/asd/vendor/silverstripe/framework/src/ORM/DataObject.php:2102
```

my phpunit.xml

```
<?xml version="1.0" encoding="UTF-8"?>
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:noNamespaceSchemaLocation="phpunit.xsd"
    bootstrap="vendor/silverstripe/framework/tests/bootstrap.php"
    backupGlobals="false"
    verbose="true">
    <testsuites>
        <testsuite name="small">
            <directory suffix=".php">app/tests</directory>
        </testsuite>
    </testsuites>

    <filter>
        <whitelist >
            <directory suffix=".php">app/controllers</directory>
            <directory suffix=".php">app/extensions</directory>
            <directory suffix=".php">app/cms</directory>
            <directory suffix=".php">app/helpers</directory>
            <directory suffix=".php">app/models</directory>
            <directory suffix=".php">app/pages</directory>
            <directory suffix=".php">app/tasks</directory>
        </whitelist>
    </filter>

    <php>
        <const name="PHPUNIT_TESTSUITE" value="true"/>
        <var name="db" value="sqlite3"/>
    </php>
</phpunit>

```

anything you can see that i have done wrong?

let me know if theres any more info you would find useful

**Comment by robbieaverill** [comment]
Yeah this is a bug in the test fixtures for annotating many many through lists. I've fixed it in #116

**Comment by phyzical** [comment]
sweet thanks!
